tikv/split_region.go

```diff
@@ -141,80 +137,48 @@ func (s *KVStore) batchSendSingleRegion(bo *Backoffer, batch kvrpc.Batch, scatte
 		}
 	}
 
-	req := tikvrpc.NewRequest(tikvrpc.CmdSplitRegion, &kvrpcpb.SplitRegionRequest{
-		SplitKeys: batch.Keys,
-	}, kvrpcpb.Context{
-		Priority: kvrpcpb.CommandPri_Normal,
-	})
-
-	sender := locate.NewRegionRequestSender(s.regionCache, s.GetTiKVClient())
-	resp, err := sender.SendReq(bo, req, batch.RegionID, client.ReadTimeoutShort)
-
-	batchResp := kvrpc.BatchResult{Response: resp}
-	if err != nil {
-		batchResp.Error = err
-		return batchResp
-	}
-	regionErr, err := resp.GetRegionError()
+	resp, err := s.pdClient.SplitRegions(bo.GetCtx(), batch.Keys)
 	if err != nil {
-		batchResp.Error = err
-		return batchResp
-	}
-	if regionErr != nil {
-		err := bo.Backoff(retry.BoRegionMiss, errors.New(regionErr.String()))
-		if err != nil {
-			batchResp.Error = err
-			return batchResp
-		}
-		resp, err = s.splitBatchRegionsReq(bo, batch.Keys, scatter, tableID)
-		batchResp.Response = resp
-		batchResp.Error = err
-		return batchResp
+		resp := &pdpb.SplitRegionsResponse{}
+		resp.Header = &pdpb.ResponseHeader{}
+		resp.Header.Error = &pdpb.Error{Message: err.Error()}
+		return resp
 	}
-
-	spResp := resp.Resp.(*kvrpcpb.SplitRegionResponse)
-	regions := spResp.GetRegions()
-	if len(regions) > 0 {
+	regionIDs := resp.GetRegionsId()
+	if len(regionIDs) > 0 {
 		// Divide a region into n, one of them may not need to be scattered,
 		// so n-1 needs to be scattered to other stores.
-		spResp.Regions = regions[:len(regions)-1]
-	}
-	var newRegionLeft string
-	if len(spResp.Regions) > 0 {
-		newRegionLeft = logutil.Hex(spResp.Regions[0]).String()
+		regionIDs = regionIDs[:len(regionIDs)-1]
 	}
+
 	logutil.BgLogger().Info("batch split regions complete",
 		zap.Uint64("batch region ID", batch.RegionID.GetID()),
 		zap.String("first at", kv.StrKey(batch.Keys[0])),
-		zap.String("first new region left", newRegionLeft),
-		zap.Int("new region count", len(spResp.Regions)))
-
+		zap.Int("new region count", len(regionIDs)))
+	if resp.FinishedPercentage != 100 {
+		err = errors.Errorf("Fail to batch split regions, finishedPercentage : %d, batch region ID : %d",
```

Comment:
where is the `err` be used?

Reply:
I forget to use before,I want to use it like this
```
	if resp.FinishedPercentage != 100 {
		err = errors.Errorf("Fail to batch split regions, finishedPercentage : %d, batch region ID : %d",
			resp.FinishedPercentage, batch.RegionID.GetID())
		resp.Header.Error = &pdpb.Error{Message: err.Error()}
	}
```

Reply:
> where is the `err` be used?
I forget to use before,I want to use it like this
```
	if resp.FinishedPercentage != 100 {
		err = errors.Errorf("Fail to batch split regions, finishedPercentage : %d, batch region ID : %d",
			resp.FinishedPercentage, batch.RegionID.GetID())
		resp.Header.Error = &pdpb.Error{Message: err.Error()}
	}
```